ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -292,24 +305,19 @@ namespace ix
             {
                 ssize_t ret = _socket->recv((char*)&_readbuf[0], _readbuf.size());
 
-                if (ret < 0 && (_socket->getErrno() == EWOULDBLOCK ||
-                                _socket->getErrno() == EAGAIN))
+                if (ret < 0 && _readyState != CLOSING && (_socket->getErrno() == EWOULDBLOCK ||
```

Comment:
I'd like it if we could find a way to handle that CLOSING in a different branch or place, because that conditional is too hard to understand now and will lead to bugs.

Is there a way to do so ?

Reply:
It’s likely to avoid the -1 value when the connection is already closed, it’s the case if we triggered the close and to branch in the next condition